source/extensions/network/dns_resolver/cares/dns_impl.h

```diff
@@ -113,5 +115,36 @@ class DnsResolverImpl : public DnsResolver, protected Logger::Loggable<Logger::I
   const absl::optional<std::string> resolvers_csv_;
 };
 
+class CaresDnsResolverFactoryImpl : public DnsResolverFactory {
+public:
+  std::string name() const override { return "envoy.dns_resolver.cares"; }
+
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return ProtobufTypes::MessagePtr{new envoy::config::core::v3::DnsResolutionConfig};
+  }
+
+  DnsResolverSharedPtr createDnsResolverCb(Event::Dispatcher& dispatcher,
+                                           const Api::Api&,
+                                           const envoy::config::core::v3::TypedExtensionConfig& dns_resolver_config) override {
+    envoy::config::core::v3::DnsResolutionConfig dns_resolution_config;
+    envoy::config::core::v3::DnsResolverOptions dns_resolver_options;
+    std::vector<Network::Address::InstanceConstSharedPtr> resolvers;
```

Comment:
The actual resolvers vector is being created for each resolver object that this factory creates. IIIUC, previously there was a single vector that was used by each resolver, no?

Reply:
This resolver vector is only used by cares DNS resolver. The previous code creating it for Apple DNS resolver may not be needed. This change cleaned this up.